### ISSUE TITLE
Fire data processing: Automatically update data before running reports, if needed

### ISSUE THREAD

**Issue by lbross** [PSU-CSAR/bagis-pro#52]
The fire data processing steps could take a long time so we don't want to generate the data every time we run the reports. But we do want to make sure we have the most current data available. TBD if this is optional with a checkbox on the report UI or if we always do it.

NIFC: We have established procedures for getting the most recent year from the service. As part of the NIFC data creation, we will record the max date in the BAGIS analysis.xml parameter file at the AOI level. If the service date exceeds the max date in the analysis.xml, we will regenerate the polygon layer from the service date and 30 preceding years. This could result in some of the oldest records purging off if they are outside the 30 year window.

MTBS: We have established procedures for getting the most recent year from the image service names on the server. If the server image service names are more recent than the aoi image layer names, we will clip the new layers to the aoi. Do we want to delete aoi image layers that are outside the 30 year window?

**Comment by jdduh** [comment]
For MTBS, if there is a mechanism to check the presence of previously-clipped images before reclipping the imageservices, then we can just leave the clipped images in the aoi without deleting them.

**Comment by lbross** [comment]
@jdduh: Do you want to add this mtbs mechanism to the data processing step? I think we could do this in the initial implementation without too much trouble. Do you want a checkbox to override this in case of bad data or do we want it to always work this way?

**Comment by jdduh** [comment]
Yes, please create a checkbox "Overwrite existing MTBS rasters in AOI" with it checked by default. I assume that the tool will overwrite nifc data each time the tool is executed.

**Comment by jdduh** [comment]
Please add checkboxes to separate the clipping of MTBS and NIFC data. MTBS is updated annually or longer, but NIFC (current) is updated "real time". It's very likely (99%) that we will need to update the NIFC data before releasing the fire disturbance statistics for the final delivery. Please add two checkboxes, "MTBS" and "NIFC", to the "Retrieve AOI Fire Data" with the default value "Checked" for both. Keep the "Overwrite existing MTBS rasters in AOI" unchecked, but disabled unless the MTBS option is checked. This combination allows users to select which source data to be retrieved to the AOIs.

**Comment by jdduh** [comment]
Looks good. Hopefully it doesn't add too much trouble to update the fire_analysis.json file when we clip NIFC and MTBS separately. I just noticed that the fire_analysis.json from the AOIs that failed to have NIFC clipped has some analysis metadata missing. Here is an example.

[fire_analysis.json](https://github.com/user-attachments/files/19983685/fire_analysis.json)

**Comment by lbross** [comment]
Good catch! The original implementation of the fire_analysis.json file assumed NIFC and MTBS would be clipped at the same time. I have fixed this.
